t/e2e/whisper/whisper_test.go

```diff
@@ -92,3 +87,122 @@ func (s *WhisperTestSuite) TestWhisperFilterRace() {
 
 	<-allFiltersAdded
 }
+
+func (s *WhisperTestSuite) TestSelectAccount() {
+	s.StartTestBackend()
+	defer s.StopTestBackend()
+
+	whisperService, err := s.Backend.NodeManager().WhisperService()
+	s.NoError(err)
+
+	// create an acc
+	address, pubKey, _, err := s.Backend.AccountManager().CreateAccount(TestConfig.Account1.Password)
+	s.NoError(err)
+
+	// make sure that identity is not (yet injected)
+	s.False(whisperService.HasKeyPair(pubKey), "identity already present in whisper")
+
+	// try selecting with wrong password
+	err = s.Backend.SelectAccount(address, "wrongPassword")
+	expectedErr := errors.New("cannot retrieve a valid key for a given account: could not decrypt key with given passphrase")
```

Comment:
feels quite fragile, i would add a an error sub-type or just check that error is received